File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        ans = newHead = ListNode()
        curr = head.next
        tot = 0
        
        while(curr):
            if curr.val is not 0:
                tot = tot + curr.val
            else:
                newHead.next = ListNode(tot)
                tot = 0
                newHead = newHead.next
            curr = curr.next
        return ans.next

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var mergeNodes = function(head) {
    let newHead = new ListNode();
    let curr = head.next;
    let ans = newHead;
    let sum = 0;
    
    while(curr){
        if(curr.val != 0){
            sum += curr.val
        } else {
            newHead.next = new ListNode(sum)
            sum = 0
            newHead = newHead.next
        }
        curr = curr.next
    }
    return ans.next
};